pkg/ebpf/compiler/compiler.cpp

```diff
@@ -11,63 +11,68 @@
-    defaultCflags({
-        "clang", // DO NOT REMOVE, first flag is ignored
-        "-emit-llvm",
-        "-O2",
-        "-D__KERNEL__",
-        "-fno-color-diagnostics",
-        "-fno-unwind-tables",
-        "-fno-asynchronous-unwind-tables",
-        "-fno-stack-protector",
-        "-nostdinc",
-        "-isystem/virtual/lib/clang/include",
-        "-x", "c"
-    }),
-    theTriple("bpf")
+enum Architecture
 {
-    std::call_once(llvmInitialized, []{
+    PPC,
+    PPCLE,
+    S390X,
+    ARM64,
+    X86
+};
+
+ClangCompiler::ClangCompiler(const char *name)
+    : llvmContext()
+    , diagOpts(new clang::DiagnosticOptions())
+    , errStream(errString)
+    , textDiagnosticPrinter(std::make_unique<clang::TextDiagnosticPrinter>(errStream, diagOpts.get()))
+    , diagnosticsEngine(new clang::DiagnosticsEngine(diagID, diagOpts, textDiagnosticPrinter.get(), false))
+    , defaultCflags({ "clang", // DO NOT REMOVE, first flag is ignored
+          "-emit-llvm",
+          "-O2",
+          "-D__KERNEL__",
+          "-fno-color-diagnostics",
+          "-fno-unwind-tables",
+          "-fno-asynchronous-unwind-tables",
+          "-fno-stack-protector",
+          "-nostdinc",
+          "-isystem/virtual/lib/clang/include",
+          "-x", "c" })
+    , targetTriple("bpf") {
+    std::call_once(llvm_initialized, [] {
         LLVMInitializeBPFTarget();
         LLVMInitializeBPFTargetMC();
         LLVMInitializeBPFTargetInfo();
         LLVMInitializeBPFAsmPrinter();
         LLVMInitializeBPFAsmParser();
 
         for (auto f : MappedFiles::files) {
-            remapped_files[f.first] = llvm::MemoryBuffer::getMemBuffer(f.second);
+            remapped_files.push_back({
+                f.path,
+                llvm::MemoryBuffer::getMemBuffer(f.content),
+            });
         }
     });
 
-    theDriver = std::make_unique<clang::driver::Driver>(
+    clangDriver = std::make_unique<clang::driver::Driver>(
         name, getArch(), *diagnosticsEngine,
-        llvm::vfs::getRealFileSystem()
-    );
+        llvm::vfs::getRealFileSystem());
 
     std::string arch = "bpf";
     std::string Error;
 
-    theTarget = llvm::TargetRegistry::lookupTarget(theTriple.getTriple(), Error);
-    if (!theTarget) {
+    llvmTarget = llvm::TargetRegistry::lookupTarget(targetTriple.getTriple(), Error);
+    if (!llvmTarget) {
         errString = "could not lookup target";
         return;
     }
 
     llvm::TargetOptions targetOptions;
-    auto RM = llvm::Optional<llvm::Reloc::Model>();
-    targetMachine = std::unique_ptr<llvm::TargetMachine>(theTarget->createTargetMachine(
-        theTriple.getTriple(), "generic", "", targetOptions, RM, llvm::None, llvm::CodeGenOpt::Aggressive));
+    targetMachine = std::unique_ptr<llvm::TargetMachine>(llvmTarget->createTargetMachine(
+        targetTriple.getTriple(), "generic", "", targetOptions, {}, llvm::None, llvm::CodeGenOpt::Aggressive));
```

Comment:
Do you know the significance of removing the `RM` argument here?

Reply:
`RM` was defined as `auto RM = llvm::Optional<llvm::Reloc::Model>();` so basically the empty optional variant. So the semantic is preserved here